Mage.Sets/src/mage/cards/b/BiggerOnTheInside.java

```diff
@@ -0,0 +1,110 @@
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.FilterPermanent;
+import mage.filter.predicate.Predicates;
+import mage.game.Game;
+import mage.players.Player;
+import mage.target.TargetPermanent;
+import mage.target.TargetPlayer;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author notgreat
+ */
+public final class BiggerOnTheInside extends CardImpl {
+
+    private static final FilterPermanent filter
+            = new FilterPermanent("artifact or land");
+
+    static {
+        filter.add(Predicates.or(
+                CardType.ARTIFACT.getPredicate(),
+                CardType.LAND.getPredicate()
+        ));
+    }
+
+    public BiggerOnTheInside(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{3}{R}{G}");
+        this.subtype.add(SubType.AURA);
+
+        // Enchant artifact or land
+        TargetPermanent auraTarget = new TargetPermanent(filter);
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.AddAbility));
+        this.addAbility(new EnchantAbility(auraTarget));
+
+        // Enchanted permanent has "{T}: Target player adds two mana of any one color. The next spell they cast this turn has cascade."
+        Ability gainedAbility = new SimpleActivatedAbility(Zone.BATTLEFIELD, new BiggerOnTheInsideEffect(), new TapSourceCost());
+        gainedAbility.addTarget(new TargetPlayer());
+        Effect effect = new GainAbilityAttachedEffect(gainedAbility, AttachmentType.AURA);
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, effect));
+    }
+
+    private BiggerOnTheInside(final BiggerOnTheInside card) {
+        super(card);
+    }
+
+    @Override
+    public BiggerOnTheInside copy() {
+        return new BiggerOnTheInside(this);
+    }
+}
+
+class BiggerOnTheInsideEffect extends OneShotEffect { //Not a mana ability since it targets
+
+    BiggerOnTheInsideEffect() {
+        super(Outcome.Benefit);
+        staticText = "Target player adds two mana of any one color. The next spell they cast this turn has cascade";
+    }
+
+    private BiggerOnTheInsideEffect(final BiggerOnTheInsideEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public BiggerOnTheInsideEffect copy() {
+        return new BiggerOnTheInsideEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(getTargetPointer().getFirst(game, source));
+        if (player == null) {
+            return false;
+        }
+        ContinuousEffect cascadeEffect = new NextSpellCastHasAbilityEffect(new CascadeAbility());
+
+        // Make a copy of the source ability with the target as controller
+        // a hack to allow use of the common class
+        Ability changedController = source.copy();
+        changedController.setControllerId(player.getId());
+        game.addEffect(cascadeEffect, changedController);
```

Comment:
Try to use TargetController logic, so original effect can use it for text gen and choices:

Places to change:
![IMG_9837](https://github.com/magefree/mage/assets/8344157/1dc9c288-5878-4d16-8afe-ecf4e358ed05)

TargetController usage example:
![IMG_9838](https://github.com/magefree/mage/assets/8344157/7b4d0c9b-cbd6-4c3d-8110-796630660819)
![IMG_9839](https://github.com/magefree/mage/assets/8344157/db7a199d-5dd3-4299-9411-ff198afc3fde)